keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -819,6 +820,42 @@ namedResources:
   - gpu-1
 ```
 
+### Optional allocation
+
+The default used by the examples above is to allocated exactly one named
+resource instance for each request. A range may be specified to request more
+than one or, if this would be acceptable, none at all. Each allocated instance
+will be listed in the allocation result, same as when no range is
+specified. That list may be empty if nothing was allocated.
+
+This example asks for all GPUs with more than 32Gi that are available:
+```
+driverRequests:
+- driverName: cards.dra.example.com
+  requests:
+  # Each entry here is a request for one resource.
+  - namedresources:
+      range:
+        minimum: 1
+      selector: |-
+        attributes.quantity["memory"].isGreaterThan(quantity("32Gi"))
+```
+
+This example allows to run a pod with a GPU when one is available and without
+one if not:
+```
+driverRequests:
+- driverName: cards.dra.example.com
+  requests:
+  # Each entry here is a request for one resource.
+  - namedresources:
+      range:
+        minimum: 0
+        maximum: 1
+      selector: |-
+        attributes.quantity["memory"].isGreaterThan(quantity("32Gi"))
+```
+
```

Comment:
This by itself does not cover more complex requests like "Give me X of A, otherwise Y of B, or as a last resort Z of C". Having three requests with `[0:X]`, `[0:Y]`, `[0:Z]` is not sufficient because then too much (X, Y, and Z for each) or not enough could get allocated (0 of each).

For that we need some scoring mechanism that looks at all possible solutions and then evaluates which one is "best". Adding such scoring is the logical next step and would also be useful for other use cases.